python/python/tests/test_async_client.py

```diff
@@ -1063,6 +1063,144 @@ async def test_linsert(self, redis_client: TRedisClient):
+
+        # Non-list destination key
+        with pytest.raises(RequestError):
+            await redis_client.lmove(key1, key3, ListDirection.LEFT, ListDirection.LEFT)
+
+    @pytest.mark.parametrize("cluster_mode", [True, False])
+    @pytest.mark.parametrize("protocol", [ProtocolVersion.RESP2, ProtocolVersion.RESP3])
+    async def test_blmove(self, redis_client: TRedisClient):
+        key1 = "{SameSlot}" + get_random_string(10)
+        key2 = "{SameSlot}" + get_random_string(10)
+
+        # Initialize the lists
+        assert await redis_client.lpush(key1, ["2", "1"]) == 2
+        assert await redis_client.lpush(key2, ["4", "3"]) == 2
+
+        # Move from LEFT to LEFT with blocking
+        assert (
+            await redis_client.blmove(
+                key1, key2, ListDirection.LEFT, ListDirection.LEFT, 0.1
+            )
+            == "1"
+        )
+        assert await redis_client.lrange(key1, 0, -1) == ["2"]
+        assert await redis_client.lrange(key2, 0, -1) == ["1", "3", "4"]
+
+        # Move from LEFT to RIGHT with blocking
+        assert (
+            await redis_client.blmove(
+                key1, key2, ListDirection.LEFT, ListDirection.RIGHT, 0.1
+            )
+            == "2"
+        )
+        assert await redis_client.lrange(key1, 0, -1) == []
+        assert await redis_client.lrange(key2, 0, -1) == ["1", "3", "4", "2"]
+
+        # Move from RIGHT to LEFT with blocking
+        assert (
+            await redis_client.blmove(
+                key2, key1, ListDirection.RIGHT, ListDirection.LEFT, 0.1
+            )
+            == "2"
+        )
+        assert await redis_client.lrange(key2, 0, -1) == ["1", "3", "4"]
+        assert await redis_client.lrange(key1, 0, -1) == ["2"]
+
+        # Move from RIGHT to RIGHT with blocking
+        assert (
+            await redis_client.blmove(
+                key2, key1, ListDirection.RIGHT, ListDirection.RIGHT, 0.1
+            )
+            == "4"
+        )
+        assert await redis_client.lrange(key2, 0, -1) == ["1", "3"]
+        assert await redis_client.lrange(key1, 0, -1) == ["2", "4"]
+
+        # Non-existing source key with blocking
+        assert (
+            await redis_client.blmove(
+                "{SameSlot}non_existing_key", key1, ListDirection.LEFT, ListDirection.LEFT, 0.1
+            )
+            is None
+        )
+
+        # Non-list source key with blocking
+        key3 = get_random_string(10)
+        assert await redis_client.set(key3, "value") == OK
+        with pytest.raises(RequestError):
+            await redis_client.blmove(
+                key3, key1, ListDirection.LEFT, ListDirection.LEFT, 0.1
+            )
+
+        # Non-list destination key with blocking
+        with pytest.raises(RequestError):
+            await redis_client.blmove(
+                key1, key3, ListDirection.LEFT, ListDirection.LEFT, 0.1
+            )
```

Comment:
Please test how timeout handled, see `bzmpop` tests for example from #1412

Reply:
Added.